### ISSUE TITLE
Build errors with Carthage and Xcode 8

### ISSUE THREAD

**Issue by tpesce** [Moya/Moya#629]
I ran into this issue initially with an existing project that had gone through the migration process to Swift 2.3, but I've also been able to reproduce it with a new "Single View Application" iOS project.

Carthage version: 0.18
Moya version: 7.0.2
Xcode version: 8.0 (8A218a)

After creating a new project, I add a Cartfile with the following contents:

```
github "Moya/Moya" ~> 7.0
```

Then:

```
$ carthage bootstrap --platform iOS
*** No Cartfile.resolved found, updating dependencies
*** Fetching Moya
*** Fetching RxSwift
*** Fetching Alamofire
*** Fetching Result
*** Checking out Alamofire at "3.5.0"
*** Checking out Result at "2.1.3"
*** Checking out ReactiveCocoa at "v4.2.2"
*** Checking out RxSwift at "2.6.0"
*** Checking out Moya at "7.0.2"
*** xcodebuild output can be found in /var/folders/d1/h0cf2xs12gv7w1wx5k41q7sc0000gp/T/carthage-xcodebuild.7jZFKG.log
*** Building scheme "Alamofire iOS" in Alamofire.xcworkspace
*** Building scheme "Result-iOS" in Result.xcodeproj
*** Building scheme "ReactiveCocoa-iOS" in ReactiveCocoa.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace'.
*** Building scheme "RxBlocking-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxCocoa-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxSwift-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxTests-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/tim/tmp/MoyaTest/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "Moya iOS" in Moya.xcodeproj
** CLEAN FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
/Users/tim/tmp/MoyaTest/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/Swift/SignalProducer.swift:1945:69: warning: 'buffer' is deprecated: Use properties instead. 'buffer' will be removed in RAC 5.0
A shell task (/usr/bin/xcrun xcodebuild -project /Users/tim/tmp/MoyaTest/Carthage/Checkouts/Moya/Moya.xcodeproj -scheme "Moya iOS" -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** CLEAN FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
```

The end of the build log shows what I think is the root issue:

```
/usr/bin/xcrun xcodebuild -project /Users/tim/tmp/MoyaTest/Carthage/Checkouts/Moya/Moya.xcodeproj -scheme "Moya iOS" -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean buildBuild settings from command line:
    BITCODE_GENERATION_MODE = bitcode
    CARTHAGE = YES
    CODE_SIGN_IDENTITY = 
    CODE_SIGNING_REQUIRED = NO
    ONLY_ACTIVE_ARCH = NO
    SDKROOT = iphoneos10.0

=== CLEAN TARGET Moya iOS OF PROJECT Moya WITH CONFIGURATION Release ===

Check dependencies
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.

=== BUILD TARGET Moya iOS OF PROJECT Moya WITH CONFIGURATION Release ===

Check dependencies
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.
```

**Comment by ashfurrow** [comment]
Sorry you're running into this error, it's possible I made a mistake when releasing 7.0.2 for Swift 2.3. Could someone from @Moya/contributors who is familiar with Carthage take a look and see if 7.0.2, the latest release, works with Xcode 8/Swift 2.3?

**Comment by tpesce** [comment]
No worries, I know you are all working hard on the recent Swift/Xcode changes.

If I had to guess, it looks like the migration changes to the Moya project file weren't included in the commit with the other Swift 2.3 updates #(2fc967f).

**Comment by ashfurrow** [comment]
Ah, interesting! Would you be able to submit a PR against the release-7.0.1-take-two branch? Don't worry about it, though, I can take a crack tomorrow.

**Comment by tpesce** [comment]
I've mostly got a PR together; just finishing up some testing. Expect to see it within a few hours.

**Comment by ashfurrow** [comment]
Wonderful, thanks!

**Comment by tpesce** [comment]
This is fixed with #636.
